R/pkg/inst/tests/testthat/test_sparkSQL.R

```diff
@@ -196,6 +196,12 @@ test_that("create DataFrame from RDD", {
   expect_equal(dtypes(df), list(c("name", "string"), c("age", "int"), c("height", "float")))
   expect_equal(as.list(collect(where(df, df$name == "John"))),
                list(name = "John", age = 19L, height = 176.5))
+  expect_equal(getNumPartitions(toRDD(df)), 1)
```

Comment:
And so we this subtlety is significant we could change to this. It's a slightly more involved change but it would match Scala exactly.

```
splits <-
unlist(lapply(0: (numSlices - 1), function(x) {
   start <- trunc((x * length)/numSlices)
   end <- trunc(((x + 1) * length)/numSlices)
   rep(start, end - start)
}))
```

And you get this sequence for length <- 50, numSlices <- 22
```
 [1]  0  0  2  2  4  4  6  6  6  9  9 11 11 13 13 15 15 15 18 18 20 20 22 22 22
[26] 25 25 27 27 29 29 31 31 31 34 34 36 36 38 38 40 40 40 43 43 45 45 47 47 47
```

For calling split() with this sequence is used as.factor - so the numeric values are not significant